src/cls/IPM/Utils/LegacyCompat.cls

```diff
@@ -0,0 +1,34 @@
+Include %IPM.Formatting
+
+Class %IPM.Utils.LegacyCompat
+{
+
+/// This method updates the superclasses of a list of classes that have been loaded but not compiled.
+ClassMethod UpdateSuperclassAndCompile(ByRef tClasses, ByRef qSpec = "") As %Status
+{
+    Set tName = "" 
+    Set tToCompile = ""
+    For {
+        Set tName = $ORDER(tClasses(tName))
+        If (tName = "") { Quit } 
+        If ($$$lcase($EXTRACT(tName, *-3, *)) '= ".cls") { Continue }
+
+        Set tClassName = $EXTRACT(tName, 1, *-4)
+        Set tToCompile(tClassName) = ""
+        Set tClass = ##class(%Dictionary.ClassDefinition).%OpenId(tClassName)
```

Comment:
Nit: Should check for super class using `$$$defClassKeyGet(tClassname,$$$cCLASSsuper)` to not load the full class definition into memory unless we need to change something.

Reply:
Done! Learned something new again